executor/insert_test.go

```diff
@@ -1576,3 +1576,14 @@ func (s *testSuite10) TestBinaryLiteralInsertToSet(c *C) {
 	tk.MustExec("insert into bintest(h) values(0x61)")
 	tk.MustQuery("select * from bintest").Check(testkit.Rows("a"))
 }
+
+func (s *testSuite10) TestStringtoDecimal(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (id decimal(10))")
+	tk.MustGetErrCode("insert into t values('1sdf')", errno.ErrTruncatedWrongValueForField)
```

Comment:
we can add more tests here to simplify the coverage.

eg: for the line you remove in L424, the '12e-3', '12A' should be added.

Reply:
Done.  PTAL~   thx~